### PR TITLE
Exposes ListTile.shape for CheckboxListTile and SwitchListTile

### LINKED ISSUE #66982: SwitchListTile and CheckboxListTile cannot be styled as a ListTile, missing shape parameter
## Trying to round corners of the SwitchTileList and CheckboxListTile  

There is a missing parameter for adding shape. I can't make the tiles in a Drawer look the same.
I also tried to wrap the whole SwitchListTile with ClipRRect without success,

## Proposal

I would imagine that exposing the shape parameter of ListTile would do the job. Most of the props are already forwarded to ListTile like dense, selected, contentPadding etc.

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Expose ListTile.shape in CheckboxListTile and SwitchListTile

## Related Issues

Fixes #66982

## Tests

I added the following tests:

- Added tests to check CheckboxListTile and SwitchListTile respects shape.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/material/checkbox_list_tile_test.dart [hunk 1]
```diff
@@ -224,4 +224,21 @@ void main() {
     await tester.pumpAndSettle();
     expect(_value, false);
   });
+
+  testWidgets('CheckboxListTile respects shape', (WidgetTester tester) async {
+    const ShapeBorder shapeBorder = RoundedRectangleBorder(
+        borderRadius: BorderRadius.horizontal(right: Radius.circular(100))
```
Comment (shihaohong) [inline packages/flutter/test/material/checkbox_list_tile_test.dart:230]: nit:
```suggestion
      borderRadius: BorderRadius.horizontal(right: Radius.circular(100)),
```

File: packages/flutter/test/material/switch_list_tile_test.dart [hunk 2]
```diff
@@ -340,4 +340,23 @@ void main() {
     expect(listTile.leading.runtimeType, Icon);
     expect(listTile.trailing.runtimeType, Switch);
   });
+
+  testWidgets('SwitchListTile respects shape', (WidgetTester tester) async {
+    const ShapeBorder shapeBorder = RoundedRectangleBorder(
+        borderRadius: BorderRadius.horizontal(right: Radius.circular(100))
```
Comment (shihaohong) [inline packages/flutter/test/material/switch_list_tile_test.dart:346]: nit: 
```suggestion
      borderRadius: BorderRadius.horizontal(right: Radius.circular(100)),
```

**Review by shihaohong — APPROVED** [review]
LGTM, just some nits for indenting

**Comment by AyushBherwani1998** [comment]
Thanks, done :)

**Pull request merged**